src/ophyd_async/core/detector.py

```diff
@@ -73,17 +83,31 @@ async def arm(
         trigger: DetectorTrigger = DetectorTrigger.internal,
         exposure: Optional[float] = None,
     ) -> AsyncStatus:
-        """Arm the detector and return AsyncStatus.
+        """Arm detector, do all necessary steps to prepare detector for triggers.
 
-        Awaiting the return value will wait for num frames to be written.
+        Args:
+            num: Expected number of frames
```

Comment:
Any idea what the user should do if the number of frames is unknown upfront? Would be good to include something about that here

Reply:
I think we should support `0` as "go forever", but detectors that don't support it should raise

Reply:
Hmm, how about `-1`? A more traditional stand-in for infinity.
Although I would still rather two separate APIs if that's possible.

Reply:
Continued in #162